components/progress-bar/index.jsx

```diff
@@ -0,0 +1,163 @@
+import classNames from 'classnames';
+
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+import { PROGRESS_BAR } from '../../utilities/constants';
+
+
+const propTypes = {
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * CSS classes to be added to tag with `.slds-progress-bar`. Uses `classNames` [API](https://github.com/JedWatson/classnames).
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * Label for the progress bar
+	 */
+	label: PropTypes.string,
+	/**
+	 *  Set radius of progress bar
+	 */
+	radius: PropTypes.oneOf(['circular']),
+	/**
+	 *  Set fill of progress bar
+	 */
+	color: PropTypes.oneOf(['success']),
+	/**
+	 *  Set progress bar thickness
+	 */
+	thickness: PropTypes.oneOf(['x-small', 'small', 'medium', 'large']),
+	/**
+	 * Percentage of progress completion, ranging [0, 100].
+	 */
+	value: PropTypes.number.isRequired,
+};
+
+const defaultProps = {};
+
+/**
+ * A progress bar component communicates to the user the progress of a particular process
+ */
+class ProgressBar extends React.Component {
+	componentWillMount() {
+		this.generatedId = shortid.generate();
+	}
+	/**
+	 * ID as a string
+	 * @returns {string} id
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	/**
+	 * Border Radius as String
+	 * @returns {string} Border
+	 */
+	getRadius() {
+		if (this.props.radius) {
+			return `slds-progress-bar_${this.props.radius}`;
+		}
+		return '';
+	}
+
+	/**
+	 * Fill color as a string
+	 * @returns {string} Color
+	 */
+	getColor() {
+		if (this.props.color) {
+			return `slds-progress-bar__value_${this.props.color}`;
+		}
+		return '';
+	}
+
+	/**
+	 * Bar thickness as a string
+	 * @returns {string} Thickness
+	 */
+	getThickness() {
+		if (this.props.thickness) {
+			return `slds-progress-bar_${this.props.thickness}`;
+		}
+		return 'slds-progress-bar_medium';
+	}
+
+	/**
+	 * Enables Descriptive Progress Bar if label is provided
+	 * @returns {string} description
+	 */
+	getDescription() {
+		if (this.props.label) {
+			return (
+				<div
+					className="slds-grid slds-grid_align-spread slds-p-bottom_x-small"
+					id="progress-bar-label-id-4"
+				>
+					<span>{this.props.label}</span>
+					<span aria-hidden="true">
+						<strong>{this.progressPercent()} Complete</strong>
+					</span>
+				</div>
+			);
+		}
+		return '';
+	}
+
+	/**
+	 * Progress percentage as a string
+	 * @returns {string} Percentage
+	 */
+	progressPercent() {
+		return `${this.props.value}%`;
+	}
+
+	render() {
+		return (
+			<div id={this.getId()}>
+				{this.getDescription()}
+				<div
+					className={classNames(
+						'slds-progress-bar',
+						this.getRadius(),
+						this.getThickness(),
+						this.props.className
+					)}
+					style={{
+						width: this.progressPercent(),
```

Comment:
Can't we just do it inline here instead of using a function?

Reply:
:+1: